toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Records.jsm

```diff
@@ -76,8 +76,11 @@ class FfiConverterOptional{{ record.nm() }} {
         return buf;
     }
 
-    static computeSize() {
-        return 1 + FfiConverter{{record.nm()}}.computeSize();
+    static computeSize(value) {
+        if (!value) {
+            return 1;
+        }
+        return 1 + FfiConverter{{record.nm()}}.computeSize(value);
```

Comment:
For other types we use `type_.ffi_converter_name()`, but I guess you can't do that here because you have a `Record` instance rather than `Type::Record`.  I'm going to try to push that code I was mentioning yesterday, I think it could help here.